### PR TITLE
Fix pytest error following deprecation

### PR DESCRIPTION
**Pull request by EmilyBourne**
Fix the pytest error during test collection for the `ndarrays/` tests following the deprecation of certain parameters in pytest. The minimum version of pytest is increased to reflect this change

### CODE REVIEW

File: tests/ndarrays/conftest.py [hunk 1]
```diff
@@ -1,51 +1,43 @@
...
-    """
-    if path.ext == ".c" and path.basename.startswith("test_"):
-        if NEEDS_FROM_PARENT:
-            return CTestFile.from_parent(path=pathlib.Path(path), parent=parent)
-        return CTestFile(parent=parent, path=pathlib.Path(path))
-
-def pytest_collection_modifyitems(items):
-    """
-    a hook to modify the items before the tests for C unit test files.
+    A hook to collect test_*.c test files.
```
Comment (bauom) [inline tests/ndarrays/conftest.py:13]: duplicated line.

File: tests/ndarrays/conftest.py [hunk 2]
```diff
@@ -1,51 +1,43 @@
...
-            item._nodeid = item.nodeid + " < " + item.test_result["DSCR"] + " >"
+    if file_path.suffix == ".c" and file_path.name.startswith("test"):
+        return CTestFile.from_parent(path=pathlib.Path(file_path), parent=parent)
+    return None
 
 class CTestFile(pytest.File):
     """
     A custom file handler class for C unit test files.
 
+    A custom file handler class for C unit test files.
```
Comment (bauom) [inline tests/ndarrays/conftest.py:30]: duplicated line.

**Review by bauom — CHANGES_REQUESTED** [review]

File: tests/ndarrays/conftest.py (hunk 1 above)
Comment (EmilyBourne) [reply tests/ndarrays/conftest.py:13]: Yes. I assumed that the documentation tests would be run on this file so I tried to write "extended descriptions". It seems it is not parsed though so I can remove them

Comment (EmilyBourne) [reply tests/ndarrays/conftest.py]: Fixed ea0fd4c2

File: tests/ndarrays/conftest.py (hunk 2 above)
Comment (EmilyBourne) [reply tests/ndarrays/conftest.py]: Fixed ea0fd4c2

**Review by bauom — APPROVED** [review]

**Pull request merged**